Mage.Sets/src/mage/cards/c/Celebr8000.java

```diff
@@ -0,0 +1,117 @@
+        // For each other result, it gains the indicated ability until end of turn.
+        // If you rolled doubles, it also gains double strike until end of turn.
+        // * 2 -- menace
+        // * 3 -- vigilance
+        // * 4 -- lifelink
+        // * 5 -- flying
+        // * 6 -- indestructible
+        this.addAbility(
+                new BeginningOfCombatTriggeredAbility(new Celebr8000Effect(), TargetController.YOU, false));
+    }
+
+    private Celebr8000(final Celebr8000 card) {
+        super(card);
+    }
+
+    @Override
+    public Celebr8000 copy() {
+        return new Celebr8000(this);
+    }
+}
+
+class Celebr8000Effect extends OneShotEffect {
+
+    private static final ContinuousEffect[] d6_effects = {
+            /* 0 */null,
+            /* 1 */new BoostSourceEffect(1, 1, Duration.EndOfTurn),
+            /* 2 */new GainAbilitySourceEffect(new MenaceAbility(), Duration.EndOfTurn),
+            /* 3 */new GainAbilitySourceEffect(VigilanceAbility.getInstance(), Duration.EndOfTurn),
+            /* 4 */new GainAbilitySourceEffect(LifelinkAbility.getInstance(), Duration.EndOfTurn),
+            /* 5 */new GainAbilitySourceEffect(FlyingAbility.getInstance(), Duration.EndOfTurn),
+            /* 6 */new GainAbilitySourceEffect(IndestructibleAbility.getInstance(), Duration.EndOfTurn)
+    };
+
+    private static final ContinuousEffect doubles_effect =
+            new GainAbilitySourceEffect(DoubleStrikeAbility.getInstance(), Duration.EndOfTurn);
+
+    Celebr8000Effect() {
+        super(Outcome.Benefit);
+        setText("roll two six-sided dice. For each result of 1, Celebr-8000 gets +1/+1 until end of turn. " +
+                "For each other result, it gains the indicated ability until end of turn. " +
+                "If you rolled doubles, it also gains double strike until end of turn.<br>" +
+                "• 2 — menace<br>" +
+                "• 3 — vigilance<br>" +
+                "• 4 — lifelink<br>" +
+                "• 5 — flying<br>" +
+                "• 6 — indestructible");
+    }
+
+    private Celebr8000Effect(final Celebr8000Effect effect) {
+        super(effect);
+    }
+
+    @Override
+    public Celebr8000Effect copy() {
+        return new Celebr8000Effect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        List<Integer> results = player.rollDice(outcome, source, game, 6, 2, 0);
+
+        int roll1 = results.get(0);
+        int roll2 = results.get(1);
+
+        if (roll1 >= 1 && roll1 <= 6) {
+            game.addEffect(d6_effects[roll1], source);
+        }
+        if (roll2 >= 1 && roll2 <= 6) {
+            game.addEffect(d6_effects[roll2], source);
+        }
+        if (roll1 == roll2){
+            game.addEffect(doubles_effect, source);
```

Comment:
Yep - `apply()` for `OneShotEffect`s - this is the way to do it for continuous effects. If you call `apply()` just once for a continuous effect then it gets reset with the next thing that happens in the game state.

I was referring to the fact that you declare the effects `static` which is slightly unusual, so I had to go look to make sure it would work fine.